docs/knowledgebase/runtime/macros.md

```diff
@@ -2,58 +2,268 @@
+  - `Call<T: Trait>` implements `frame_support::dispatch::GetDispatchInfo` and `frame_support::dispatch::Dispatchable` traits that contain the core logic for dispatching calls.
+  - `Module<T>` implements `frame_support::dispatch::Callable<T>` trait with its associated type `Call` assigned to the `Call` enum.
+  - Each dispatchable function is prepended with internal tracing logic for keeping track of the node activities. Developers can set whether to trace for a dispatchable function by specifying its interest level.
+
+### construct_runtime!
+
+**When to Use**
+
+To construct Substrate runtime and integrating various pallets into the runtime.
+
+**What It Does**
+
+The macro declares and implements various struct and enum, e.g.`Runtime`, `Event`, `Origin`, `Call`, `GenesisConfig` etc, and implements various helper traits for these struct types. The macro also adds logics of mapping different events and dispatchable calls from the overarching runtime back to a particular pallet.
+
+**Doc. and Example**
+
+  - [API Documentation](https://substrate.dev/rustdocs/v2.0.0-rc3/frame_support/macro.construct_runtime.html)
+  - [Macro Definition](https://github.com/paritytech/substrate/blob/v2.0.0-rc3/frame/support/procedural/src/construct_runtime/mod.rs#L31-L36)
+  - Macro Expansion Example: [original](https://gist.github.com/jimmychu0807/c4a88ec8e0342ee9f4e14bd26287324e#file-runtime-lib-rs-L260-L277), [expanded](https://gist.github.com/jimmychu0807/c4a88ec8e0342ee9f4e14bd26287324e#file-runtime-lib-expanded-rs-L763-L2475)
+
+**Macro Implementation Notes**
+
+  - `Runtime` struct type is defined to represent the Substrate runtime.
+  - `Event` enum type is defined with variants of all pallets that emit events, with helper traits and encoding/decoding traits implemented for the enum. Various conversion traits `Event` also implements `TryInto<pallets::Event<Runtime>>` trait to extract the event out from the enum type.
+  - `Origin` enum type is defined with helper traits, e.g. `PartialEq`, `Clone`, `Debug` implemented. This enum type defines who call an extrinsic, `NONE`, `ROOT`, or signed by a particular account.
+  - `Call` enum type is defined with all integrated pallets as variants. It contains the data and metadata of each of the integrated pallets, and redirects calls to the specific pallet via implementing `frame_support::dispatch::Dispatchable` trait.
+  - `GenesisConfig` struct type is defined and implements `sp_runtime::BuildStorage` trait for building up the storage genesis config.
+  - The macro provides a default `frame_support::unsigned::ValidateUnsigned` trait implementation if not provided to disallow all unsigned transactions.
```

Comment:
it is implementing ValidateUnsigned using each module ValidateUnsigned implementation, indeed if no module has any ValidateUnsigned specified then no unsigned transactions are allowed.

I just feel a bit weird saying this is `a default` implementation, it is just `the` implementation IMO

Reply:
I like the rephrasing 👍